<launch>
  <!-- Connect rosserial to Teensy -->
  <node name="hand_serial_com" pkg="rosserial_python" type="serial_node.py">
    <param name="port" value="/dev/ttyACM3"/>
    <param name="baud" value="115200"/>
  </node>

  <!-- Start joy_node to connect to controller -->
  <node name="joy_node" pkg="joy" type="joy_node" output="log" respawn="false">
    <param name="dev" value="/dev/input/js0" />
  </node>

  <!-- Run hand_teleop.py script -->
  <node name="hand_teleop" pkg="hand_arduino" type="hand_teleop.py">
  </node>

  <!-- Open URDF in rviz -->
  <arg
    name="model" />
  <param
    name="robot_description"
    textfile="$(find hand_description)/urdf/underactuated_hand_urdf.urdf" />
  <node
    name="joint_state_publisher_gui"
    pkg="joint_state_publisher_gui"
    type="joint_state_publisher_gui" >
    <rosparam param="source_list">["/joint_states_command"]</rosparam>
  </node>
  <node
    name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="robot_state_publisher" />
  <node
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(find hand_rviz)/rviz/teleop_ft_config.rviz" />
  <rosparam file="$(find hand_rviz)/config/pot_ranges.yaml" />
  <node
    name="pot_mins"
    pkg="hand_rviz"
    type="server.py" />

  <!-- Run onrobot_hex_publisher_wrench.py script -->
  <node name="force_torque" pkg="onrobot_hex_udp_tcp" type="onrobot_hex_publisher_wrench.py">
  </node>

</launch>
